zerver/lib/markdown/__init__.py

```diff
@@ -646,6 +646,8 @@ def add_embed(self, root: Element, link: str, extracted_data: Dict[str, Any]) ->
             parsed_url = urllib.parse.urlparse(link)
             domain = '{url.scheme}://{url.netloc}/'.format(url=parsed_url)
             img_link = urllib.parse.urljoin(domain, img_link)
+
+        img_link = "/thumbnail?url={}&size=thumbnail".format(img_link)
```

Comment:
URL parameters need to be encoded: `urlunsplit(("", "", "/thumbnail", urlencode({"url": img_link, "size": "thumbnail"}), ""))`

Reply:
Also doesn’t this need to be conditional on `settings.THUMBNAIL_IMAGES`?

Reply:
Yeah, this should use the same logic that uses camo or thumbor depending on configuration.

Reply:
Resolved, thanks.